### PR TITLE
Add docker support

### PR DESCRIPTION
**Pull request by hermanjunge**
- Have already the image for `1.0.2` at @consenSys [docker hub account](https://hub.docker.com/r/consensysllc/truffle/).
- Regarding the point above: It'd be nice to add somewhere an automated task to build the image on new versions or pushes to `master`.
- A new directory `docker` had to be made, as two files were needed: A `Dockerfile` and a `entrypoint.sh` file.
- This solves #114

### CODE REVIEW

**Comment by tcoulter** [comment]
Hey @hermanjunge, can you take a look at what the String team is doing here? https://github.com/dominicwilliams/instant-dapp-ide

It's a full dapp development docker image with TestRPC and Truffle. Would love to get your opinion. I'd like to evaluate supporting their effort as that pushes maintenance of docker images over to them.

**Comment by hermanjunge** [comment]
**Done the evaluation**

Cons first, as I don't want to give the idea that `I don't want other people to do the docker stuff`. Actually I love the ole' `Give a man a fish` / `Teach him to fish` saying. That way, we have more bandwith to learn more things and solve other problems. So, here is the thing:

**Cons:**

* Running more than one process inside a container is an _anti-pattern_.
* They need help to make the image lighter.
* Forces the dev to develop using Cloud9. What if the dev just wants truffle without anything (they should not be using testrpc, but a real node)? What if they use _vim_ (or, god forbid me, _emacs_), _atom_, _sublime_, etc?
* Regarding the above: A much better pattern is an array of containers orchestrated using _docker-compose_.
* This is nitpicking, but they need help in the design of the _Dockerfile_, every line is a layer, and layers must be compact.
* Finally, they should rely more on external files (and directories), they could be easily updated at build time, adding the right instruction to the build script, in order to keep a smaller _Dockerfile_.

**Pros:**

* We won't have, in the future, to maintain anything about docker. (**HUGE PRO**)
* We can support their effort to modularize their dev environment, should they _want our help_, of course. (See above what I think of fishes and code).

**TLDR:** Cool. If they let us help them, we can have this excellent dev env: powerful slim, fit, adaptable. And we can focus on solving other problems.

**Comment by tcoulter** [comment]
Which would you want to do? I just spent a few days with them, and I'm sure
they'd be very open to us helping. I think if we do it right we can make it
a team effort.

Totally understand both your pros and cons. Good points all around.

**Comment by hermanjunge** [comment]
I can work with them, my biggest complain is that the development environment must be modularised into several containers orchestrated using _docker compose_.

**Comment by tcoulter** [comment]
I'm sure they'd love your input. Let me reach out to them to see if they're
interested and we'll go from there.

**Comment by hermanjunge** [comment]
Good. @tcoulter, is your call to keep this PR open or closed. I got this very implementation in my `dockerfiles` collection :-)

**Comment by oluies** [comment]
Herman, I am using this here https://github.com/combient/truffle
I guess you havent progresed with this somewhere else?

**Comment by hermanjunge** [comment]
@oluies Here https://github.com/hermanjunge/dockerfiles/blob/3d6b00279f044505b576560bede0b7836e4e0133/truffle/Dockerfile
